### PR TITLE
ARROW-1982: [Python] Coerce Parquet statistics as bytes to more useful Python scalar types

### PR DESCRIPTION
**Pull request by wesm**
I also changed the BYTE_ARRAY, FIXED_LEN_BYTE_ARRAY to return bytes since decoding from binary to UTF8 unicode didn't seem correct to me as the default behavior

### CODE REVIEW

File: python/pyarrow/_parquet.pyx [hunk 1]
```diff
@@ -70,6 +70,28 @@ cdef class RowGroupStatistics:
...
+    cdef inline _cast_statistic(self, object value):
+        cdef ParquetType physical_type = self.statistics.get().physical_type()
+        if physical_type == ParquetType_BOOLEAN:
+            return bool(int(value))
+        elif physical_type == ParquetType_INT32:
+            return int(value)
+        elif physical_type == ParquetType_INT64:
+            return int(value)
+        elif physical_type == ParquetType_INT96:
+            # TODO
```
Comment (xhochy) [inline python/pyarrow/_parquet.pyx]: We should return also `bytes` here.

**Review by xhochy — APPROVED** [review]
+1, LGTM

File: python/pyarrow/_parquet.pyx (hunk 1 above)
Comment (wesm) [reply python/pyarrow/_parquet.pyx]: OK, value is bytes here already so this can remain as, I'll remove the TODO

**Comment by wesm** [comment]
Appveyor build: https://ci.appveyor.com/project/wesm/arrow/build/1.0.1737

**Pull request closed without merging**